src/components/AuthForm.js

```diff
@@ -0,0 +1,142 @@
+import {
+  Form,
+  Link,
+  useSearchParams,
+  useActionData,
+  useNavigation,
+  useNavigate,
+} from "react-router-dom";
+
+import classes from "./AuthForm.module.css";
+import { useState } from "react";
+
+function AuthForm() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const isLogin = searchParams.get("mode") === "login";
+  const [otp, setOtp] = useState(false);
+  const navigation = useNavigation();
+  const isSubmitting = navigation.state === "submitting";
+  const data = useActionData();
+  const navigate = useNavigate();
+
+  async function submitHandler(event) {
+    event.preventDefault();
+    if (!otp) {
+      const data = new FormData(event.target);
+      var object = {};
+      data.forEach((value, key) => (object[key] = value));
+      var body = JSON.stringify(object);
+
+      let url = "http://127.0.0.1:8000/user/";
+
+      if (isLogin) {
+        url += "login/";
+      } else {
+        url += "register_user/";
+      }
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: body,
+      });
+
+      if (!response.ok) {
+        return response;
+      }
+
+      const resData = await response.json();
```

Comment:
Please define all of these as methods in service.js file and then use them here.

Reply:
you mean like defining a fetch method in the service,js file which will take a url, headers, method and body and return a json response when called

Reply:
yes. and if an api call is need multiple times then create a hook for that.

Reply:
Ok